pkg/codegen/codegen.go

```diff
@@ -103,7 +106,7 @@ func constructImportMapping(importMapping map[string]string) importMap {
 // Generate uses the Go templating engine to generate all of our server wrappers from
 // the descriptions we've built up above from the schema objects.
 // opts defines
-func Generate(spec *openapi3.T, opts Configuration) (string, error) {
+func Generate(spec *libopenapi.DocumentModel[v3.Document], opts Configuration) (string, error) {
```

Comment:
The `codegen` package is used by external users, so this would be a breaking change - thoughts on how we could look at making this a non-breaking change? Perhaps introducing a new package i.e. `pkg/codegen/openapi`?

Reply:
I changed the code to not to modify anything under pkg/* directories, instead created a new package under `pkg/codegen/openapi`